src/main/java/org/example/controller/DepartmentController.java

```diff
@@ -0,0 +1,42 @@
+package org.example.controller;
+
+import org.example.model.Employee;
+import org.example.service.DepartmentService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/department")
+public class DepartmentController {
+
+    private final DepartmentService departmentService;
+
+    public DepartmentController(DepartmentService departmentService) {
+        this.departmentService = departmentService;
+    }
+
+    @GetMapping("/max-salary")
+    public Employee getEmployeeWithMaxSalary(@RequestParam("departmentID") int department) {
+        return departmentService.getEmployeeWithMaxSalary(department);
+    }
+
+    @GetMapping("/min-salary")
+    public Employee getEmployeeWithMinSalary(@RequestParam("departmentID") int department) {
+        return departmentService.getEmployeeWithMinSalary(department);
+    }
+
+    @GetMapping(value = "/all", params = "departmentId")
```

Comment:
```suggestion
    @GetMapping(value = "/all", params = "departmentID")
```
параметр ниже называется `departmentID`, а не `departmentId`